### PR TITLE
Remove sleep() for universal robot

### PR DESCRIPTION
**Pull request by BenjaminHug8**
**Description**
This PR is to solved the time to time failing of the `universal_robot` system test in the CI.

**Related Issues**
This pull-request fixes issue #310

### CODE REVIEW

**Comment by BenjaminHug8** [comment]
Open this [issue](https://github.com/ros-controls/ros2_controllers/issues/265).

**Comment by BenjaminHug8** [comment]
The failing of `galactic testing` is not related to this PR, see this other [PR](https://github.com/cyberbotics/webots_ros2/pull/337).

**Comment by lukicdarkoo** [comment]
It seems to be working with the `--no-rendering` flag. I let it run again to check. Galactic and Foxy testing are failing due to other reasons.

File: webots_ros2_universal_robot/webots_ros2_universal_robot/follow_joint_trajectory_client.py [hunk 1]
```diff
@@ -56,9 +58,12 @@ def __on_get_result_callback(self, future):
             rclpy.shutdown()
 
     def send_goal(self, trajectory, iteration=1):
-        self.get_logger().info('Waiting for action server...')
+        self.get_logger().info('Waiting for action server to be ready...')
         self.__client.wait_for_server()
 
+        sleep_time = 0.25
+        time.sleep(sleep_time)
```
Comment (BenjaminHug8) [inline webots_ros2_universal_robot/webots_ros2_universal_robot/follow_joint_trajectory_client.py:65]: @lukicdarkoo just to be sure you also have those lines in your test without rendering ?

**Comment by BenjaminHug8** [comment]
> It seems to be working with the `--no-rendering` flag. I let it run again to check. Galactic and Foxy testing are failing due to other reasons.

Yeah I have seen that `Foxy testing` had a failure on an other test (and `Galactic main` will fail until next sync).

**Comment by BenjaminHug8** [comment]
Btw would not it be a good thing if each test on the CI would have the option `--no-rendering` in order to go faster?
I do not know if it is something complicated to do.

**Comment by lukicdarkoo** [comment]
> Btw would not it be a good thing if each test on the CI would have the option `--no-rendering` in order to go faster? I do not know if it is something complicated to do.

They all have now. If the `WEBOTS_OFFSCREEN` environment is specified there is no need to render anything into the screen

**Comment by BenjaminHug8** [comment]
> They all have now. If the `WEBOTS_OFFSCREEN` environment is specified there is no need to render anything into the screen

Right I read too fast.

File: webots_ros2_universal_robot/launch/multirobot_launch.py [hunk 2]
```diff
@@ -74,27 +74,31 @@ def generate_launch_description():
...
         prefix=controller_manager_prefix,
+        namespace='abb',
         arguments=['abb_joint_trajectory_controller', '-c', 'abb/controller_manager'] + controller_manager_timeout,
     )
     abb_joint_state_broadcaster_spawner = Node(
         package='controller_manager',
         executable='spawner.py',
         output='screen',
         prefix=controller_manager_prefix,
+        namespace='abb',
```
Comment (lukicdarkoo) [inline webots_ros2_universal_robot/launch/multirobot_launch.py:101]:
```suggestion
        arguments=['ur_joint_trajectory_controller', '-c', 'ur5e/controller_manager'] + controller_manager_timeout,
    )
    ur5e_joint_state_broadcaster_spawner = Node(
        package='controller_manager',
        executable='spawner.py',
        output='screen',
        prefix=controller_manager_prefix,
        arguments=['ur_joint_state_broadcaster', '-c', 'ur5e/controller_manager'] + controller_manager_timeout,
    )
    abb_trajectory_controller_spawner = Node(
        package='controller_manager',
        executable='spawner.py',
        output='screen',
        prefix=controller_manager_prefix,
        arguments=['abb_joint_trajectory_controller', '-c', 'abb/controller_manager'] + controller_manager_timeout,
    )
    abb_joint_state_broadcaster_spawner = Node(
        package='controller_manager',
        executable='spawner.py',
        output='screen',
        prefix=controller_manager_prefix,
```
These namespaces are not needed as those are just controller spawners

File: webots_ros2_universal_robot/webots_ros2_universal_robot/follow_joint_trajectory_client.py [hunk 3]
```diff
@@ -28,6 +29,7 @@ class FollowJointTrajectoryClient(Node):
     def __init__(self, name, action_name):
         super().__init__(name)
         self.__client = ActionClient(self, FollowJointTrajectory, action_name)
+        self.__action_name = action_name
```
Comment (lukicdarkoo) [inline webots_ros2_universal_robot/webots_ros2_universal_robot/follow_joint_trajectory_client.py:32]:
```suggestion
```
Not used

File: webots_ros2_universal_robot/webots_ros2_universal_robot/follow_joint_trajectory_client.py (hunk 1 above)
Comment (lukicdarkoo) [reply webots_ros2_universal_robot/webots_ros2_universal_robot/follow_joint_trajectory_client.py:65]: Let me try without it

**Comment by BenjaminHug8** [comment]
One crash on `testSlam`, rerun the tests.

**Comment by BenjaminHug8** [comment]
By removing the sleep we still had 1 failure where the goal is sent before the controller is ready.

**Comment by lukicdarkoo** [comment]
@BenjaminHug8  It seems we have to put a small delay, but in general it looks fine now?

**Review by lukicdarkoo — APPROVED** [review]
`self.__client.wait_for_server()` is not perfect, but putting a small delay seems to be working fine. Alternatively, we can query the state:
http://wiki.ros.org/joint_trajectory_controller (`query_state` service and `state` topic)

**Comment by BenjaminHug8** [comment]
> @BenjaminHug8 It seems we have to put a small delay, but in general it looks fine now?

Yes I already used a delay to make it work, but with @omichel we wanted to get rid of that.
Isn't strange to have this delay, it should work without it if we can correctly wait for the controller to be ready before sending the goals no?

**Comment by lukicdarkoo** [comment]
@BenjaminHug8 delay deleted

**Comment by BenjaminHug8** [comment]
> @BenjaminHug8 delay deleted

Perfect, it seems all good thanks!

I just corrected a typo in comment of the code.

**Comment by BenjaminHug8** [comment]
@omichel does it sound good for you like this?

**Review by omichel — APPROVED** [review]
Yes, very good. I was waiting for the tests to pass.

**Pull request merged**